platform/ext/target/nordic_nrf/nrf5340/gcc/startup_nrf5340_s.S

```diff
@@ -244,6 +244,13 @@ Reset_Handler:
 #endif /* __STARTUP_CLEAR_BSS_MULTIPLE || __STARTUP_CLEAR_BSS */
 
     cpsid   i              /* Disable IRQs */
+
+/* Setup Vector Table Offset Register. */
+    ldr     r0, =__Vectors
+    ldr     r1, =#0xE000ED08 /* SCB->VTOR */
+    str     r0, [r1]
+    dsb     0xF
```

Comment:
Changing VTOR requires ISB as well. This is because an error may occur during fetching an instruction, and we want to make sure the new vector table will apply.

Reply:
Perhaps I'm missing something or I just don't understand it well. ISB will flush the pipeline, so all the instructions following ISB will be fetched again. Consequently, for a few instructions the fetching will be done twice, with the old and with the new vector table. If an error was to occur during both these fetchings, the corresponding fault handler from the old vector table would be executed anyway first. And if the first fetching was successful, what's the point in trying once more? I don't see a clear benefit of having ISB here and I think it might be even confusing.

Reply:
https://static.docs.arm.com/dai0321/a/DAI0321A_programming_guide_memory_barriers_for_m_profile.pdf

In general, your are right @anangl, I had a more careful look. Actually, if you see the attached application note, even DSB is not required, according to the implementation requirements :)  That's because Cortex-M always completes memory writes before exception entries.

However, all this depend on the assumption that the _old_ vector table was a "valid" one. Assume, for example that the "old" vector table was existing but "dummy", e.g. was just containing dummy fault handlers which were silently ignoring faults. We are now changing to the new vector table, and, in theory, we might want to re-fetch all instructions, to see if there were a fetching fault. :)

Reply:
Removed those DSB. They're indeed redundant for Cortex-M processors.